src/ssl/PeekingPeerConnector.cc

```diff
@@ -43,6 +43,7 @@ Ssl::PeekingPeerConnector::PeekingPeerConnector(HttpRequestPointer &aRequest,
         const auto serverBump = csd->serverBump();
         Must(serverBump);
         Must(serverBump->at(XactionStep::tlsBump3));
+        currentBumpMode = serverBump->bumpModeAtStep(XactionStep::tlsBump2);
     }
     // else the client is gone, and we cannot check the step, but must carry on
```

Comment:
It is not clear to me yet _how_ we can carry on if we do not know currentBumpMode. For example, how do we know whether to peek or stare? Also, clearly splicing is not possible (because the client is gone). Bumping may be possible, but perhaps that is _not_ what the admin wanted us to do? I am not sure this is in scope, but it smells like a bug -- perhaps we should throw whenever currentBumpMode is bumpNone here, eliminating one possible currentBumpMode value from reaching the rest of PeekingPeerConnector code?

Reply:
IT does not make sense to continue if the client gone.
What about to throw if request->clientConnectionManager is gone?
The FwdState will handle exceptions (through the FwdState::advanceDestination call).

Reply:
We can throw in the constructor, but there are other PeekingPeerConnector places where we check whether the client is still there. Logically, we should throw there as well (and add a getter method to avoid code duplication). However, I am not sure it is OK to throw in PeekingPeerConnector::initialize(), PeekingPeerConnector::noteNegotiationDone(), PeekingPeerConnector::handleServerCertificate(), and PeekingPeerConnector::serverCertificateVerified(). Some of those methods might not be protected by AsyncCall wrappers.

PeekingPeerConnector::noteNegotiationDone() already has code that reacts to client connection disappearance, but that code is not usable in the PeekingPeerConnector constructor.

And, as discussed earlier, PeekingPeerConnector should not even be asking this question!

To avoid adding a new, difficult problem to this PR scope I suggest leaving all this wrong-question code as is, but record our decision/plans (if you agree with them) by changing this comment:

```suggestion
    // else the client is gone, we cannot check the step, and will probably bail later
    // TODO: Refactor so that this job does not have to (re)check client existence!
```

Reply:
commit 7901e7ea